{
    "data_version": "4.0",
    "data_type": "CVE",
    "data_format": "MITRE",
    "CVE_data_meta": {
        "ID": "CVE-2024-6232",
        "ASSIGNER": "cna@python.org",
        "STATE": "PUBLIC"
    },
    "description": {
        "description_data": [
            {
                "lang": "eng",
                "value": "There is a MEDIUM severity vulnerability affecting CPython.\n\n\n\n\n\nRegular expressions that allowed excessive backtracking during tarfile.TarFile header parsing are vulnerable to ReDoS via specifically-crafted tar archives."
            }
        ]
    },
    "problemtype": {
        "problemtype_data": [
            {
                "description": [
                    {
                        "lang": "eng",
                        "value": "CWE-1333 Inefficient Regular Expression Complexity",
                        "cweId": "CWE-1333"
                    }
                ]
            }
        ]
    },
    "affects": {
        "vendor": {
            "vendor_data": [
                {
                    "vendor_name": "Python Software Foundation",
                    "product": {
                        "product_data": [
                            {
                                "product_name": "CPython",
                                "version": {
                                    "version_data": [
                                        {
                                            "version_affected": "<",
                                            "version_name": "0",
                                            "version_value": "3.13.0rc2"
                                        }
                                    ]
                                }
                            }
                        ]
                    }
                }
            ]
        }
    },
    "references": {
        "reference_data": [
            {
                "url": "https://github.com/python/cpython/pull/121286",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/pull/121286"
            },
            {
                "url": "https://github.com/python/cpython/issues/121285",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/issues/121285"
            },
            {
                "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/JRYFTPRHZRTLMZLWQEUHZSJXNHM4ACTY/",
                "refsource": "MISC",
                "name": "https://mail.python.org/archives/list/security-announce@python.org/thread/JRYFTPRHZRTLMZLWQEUHZSJXNHM4ACTY/"
            }
        ]
    },
    "generator": {
        "engine": "Vulnogram 0.2.0"
    },
    "source": {
        "discovery": "UNKNOWN"
    },
    "credits": [
        {
            "lang": "en",
            "value": "Elias Joakim Myllym\u00e4ki"
        },
        {
            "lang": "en",
            "value": "Seth Larson"
        },
        {
            "lang": "en",
            "value": "Seth Larson"
        },
        {
            "lang": "en",
            "value": "Gregory P. Smith"
        }
    ]
}